src/scratchOrgSettingsGenerator.ts

```diff
@@ -148,8 +143,14 @@ export default class SettingsGenerator {
     const status = (await client.subscribe()) as string;
 
     if (status !== RequestStatus.Succeeded) {
+      const componentFailures = ensureObject<{
```

Comment:
ensureObject is gonna throw if there aren't details.  Any chance that could happen?

Reply:
with the `true` change above, we'll get success/failure info, and since the deploy failed, there will be failure info